public/static/gameScene.js

```diff
@@ -447,6 +471,38 @@ class GameScene extends Phaser.Scene {
 
     }
 
+    displayMissileCount(self, id, positionX, newAmount, maxAmount, regenTime) {
+        const positionY = 575;
+        const maxDisplayTimeMs = 5000;
+        let myGroup = this.add.group();
+        self.missileCounterUIs[id] = myGroup;
+        const missileSprite = self.add.sprite(positionX - 45, positionY, 'missile').setDisplaySize(20,30).setDepth(100);
+        const amountText = self.add.text(positionX - 15, positionY, '' + newAmount + '/' + maxAmount, { fontSize: '24px' }).setTint(0xffffff).setDepth(100);
+        myGroup.add(missileSprite);
+        myGroup.add(amountText);
+        self.missileCounterUIs[id] = myGroup;
+
+        //wait until either display time is up or the next one appears.
+        let timer = 0;
+        let cleanLoop = null;
```

Comment:
Is this line necessary? Couldn't we just declare and assign cleanLoop to the setInterval function, or just call the setInterval function without assigning it to a variable at all?

Reply:
possibly, i will see if that works. we have to assign it to a variable though, otherwise there's no way to clearInterval

Reply:
you are right that the first assign to null is unnecessary. the code is better now, and there was also a similar null loop set i fixed